src/redux/books/books.js

```diff
@@ -13,28 +16,66 @@ export const remove = (index) => ({
   payload: index,
 });
 
-const initialState = [{
-  id: 1,
-  title: 'book1',
-  author: 'author1',
-},
-{
-  id: 2,
-  title: 'book2',
-  author: 'author2',
-},
-];
+export const read = (books) => ({
+  type: READ,
+  payload: books,
+});
 
 // Reducer
-const booksReducer = (state = initialState, action) => {
+const booksReducer = (state = [], action) => {
   switch (action.type) {
+    case READ:
+      return action.payload;
     case ADD:
       return [...state, action.payload];
     case REMOVE:
-      return [...state.filter((item) => item.id !== action.payload)];
+      return [...state.filter((item) => item.item_id !== action.payload)];
     default:
       return state;
   }
 };
 
+/* eslint-disable array-callback-return */
+export const recieveBooks = () => async (dispatch) => {
+  await fetch(APIURL)
+    .then((res) => res.json())
+    .then((books) => {
+      const bookList = [];
+      Object.keys(books).map((key) => {
+        bookList.push({
+          item_id: key,
+          title: books[key][0].title,
+          author: books[key][0].author,
+          category: books[key][0].category,
+        });
+      });
+      dispatch(read(bookList));
+    });
+};
```

Comment:
- [ ] As requested by the previous reviewer, Please use `createAsyncThunk` to make API calls. please read this [comment](https://github.com/Farida-Hamid/Bookstore/pull/4#discussion_r975515299)